Remove usage of unittest

The ``esasky`` and ``utils/tap`` modules are using unittest which triggers an error upstream when using python 3.6. The best and easiest way to deal with it in astroquery to change this test to be pure pytest.

```
==================================== ERRORS ====================================
_ ERROR collecting lib.linux-x86_64-3.6/astroquery/esasky/tests/test_esasky.py _
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:600: in collect
    ???
_pytest.python:454: in collect
    ???
_pytest.python:466: in makeitem
    ???
_pytest.vendored_packages.pluggy:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
_pytest.vendored_packages.pluggy:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
_pytest.vendored_packages.pluggy:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
_pytest.vendored_packages.pluggy:595: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
_pytest.vendored_packages.pluggy:249: in _wrapped_call
    wrap_controller.send(call_outcome)
E   DeprecationWarning: generator 'pytest_pycollect_makeitem' raised StopIteration
```

--- a/astroquery/dace/tests/test_dace.py
+++ b/astroquery/dace/tests/test_dace.py
@@ -1,5 +1,4 @@
 import os
-import unittest
 import json
 from astroquery.dace import Dace
 
@@ -8,24 +7,19 @@ DATA_FILES = {
 }
 
 
-class TestDaceClass(unittest.TestCase):
-    def test_transform_data_as_dict(self):
-        expected_parameter_dict = {'ccf_noise': [0.005320016177906, 0.00393390440796704, 0.0032324617496158],
-                                   'ins_name': ['CORALIE98', 'CORALIE98', 'HARPS'],
-                                   'drs_qc': [True, True, False],
-                                   'rjd': [51031, 51039, 51088],
-                                   'rv': [31300.4226771379, 31295.5671320506, 31294.3391634734],
-                                   'rv_err': [5.420218247708816, 4.0697289792344185, 3.4386352834851026]}
-        with open(self._data_path('parameter_list.json'), 'r') as file:
-            parameter_list = json.load(file)
-        parameter_dict = Dace.transform_data_as_dict(parameter_list)
-        assert parameter_dict == expected_parameter_dict
-
-    @staticmethod
-    def _data_path(filename):
-        data_dir = os.path.join(os.path.dirname(__file__), 'data')
-        return os.path.join(data_dir, filename)
+def test_transform_data_as_dict():
+    expected_parameter_dict = {'ccf_noise': [0.005320016177906, 0.00393390440796704, 0.0032324617496158],
+                               'ins_name': ['CORALIE98', 'CORALIE98', 'HARPS'],
+                               'drs_qc': [True, True, False],
+                               'rjd': [51031, 51039, 51088],
+                               'rv': [31300.4226771379, 31295.5671320506, 31294.3391634734],
+                               'rv_err': [5.420218247708816, 4.0697289792344185, 3.4386352834851026]}
+    with open(_data_path('parameter_list.json'), 'r') as file:
+        parameter_list = json.load(file)
+    parameter_dict = Dace.transform_data_as_dict(parameter_list)
+    assert parameter_dict == expected_parameter_dict
 
 
-if __name__ == "__main__":
-    unittest.main()
+def _data_path(filename):
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    return os.path.join(data_dir, filename)

--- a/astroquery/gaia/tests/test_gaiatap.py
+++ b/astroquery/gaia/tests/test_gaiatap.py
@@ -14,7 +14,6 @@ Created on 30 jun. 2016
 
 
 """
-import unittest
 import os
 from unittest.mock import patch
 
@@ -40,7 +39,7 @@ def data_path(filename):
     return os.path.join(data_dir, filename)
 
 
-class TestTap(unittest.TestCase):
+class TestTap():
 
     def test_query_object(self):
         conn_handler = DummyConnHandler()
@@ -609,8 +608,3 @@ class TestTap(unittest.TestCase):
         mock_logout.side_effect = HTTPError("Login error")
         tap.logout()
         assert (mock_logout.call_count == 3)
-
-
-if __name__ == "__main__":
-    # import sys;sys.argv = ['', 'Test.testName']
-    unittest.main()

